### PR TITLE
ttnn interface lib

### PR DESCRIPTION
**Pull request by mbezuljTT**
**Disclaimer**
As agreed target brach is **feature/ttnn_interface** as we are linking compiler against tt-metal **feature/mlir-interface** (commit 57f44681e2) that breaks offline compilation on non Linux machines. We'll be in rebase business while on this feature - for now. 

**What**
Wrapping ttnn compiler interface lib (ttnn_mlir_interface.hpp && ttnn_mlir_interface.a) at LegalGridAnalysis.cpp (under mlir::ttnn_wrapper namespace) so it can be used at `bool mock_is_output_tensor_legal_for_op(Operation *op, LayoutAttr layout)`. 

**Details** 
As mock_is_output_tensor_legal_for_op  doesnt have input tensors, for now the assumption is input layout == output layout. 
This doesn't work for matmul, so it's disabled. 

I'll meet with @odjuricicTT to discuss nitty-gritty details on where to go from here (In the following PRs).

ttnn-mlir-interface-checker is basic unit test if lib is working ok. I'll migrate this to proper unittest (more likely in the following PRs)

@derdeljanTT  will follow-up with l1 consumption integration.

### CODE REVIEW

File: lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp [hunk 1]
```diff
@@ -3,13 +3,143 @@
...
+            shape, memory_config, data_type, shape, memory_config, data_type);
+  } else if (op_name_str == "ttir.relu") {
+    is_valid =
+        ttnn::mlir_interface::does_unary_op_support_input_output_constraints(
+            "RELU", // todo agree upon mapping to ttnn::mlir_interface
+            shape, memory_config, data_type, shape, memory_config, data_type);
+  } else {
+    llvm::outs() << op->getName() << " missing ttnn interface\n";
+    return false;
+  }
```
Comment (derdeljanTT) [inline lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp:140]: Is there a nicer way of doing op class dedution - in TTIR there are base classes for each op category (e.g., `TTIR_ElementwiseBinaryOp`, `TTIR_ElementwiseUnaryOp`, ...), can we use that info to get the op class?

**Review by derdeljanTT — APPROVED** [review]

File: lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp (hunk 1 above)
Comment (odjuricicTT) [reply lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp:140]: It is possible to do `llvm::isa<MatmulOp>(op)`.

Comment (mbezuljTT) [reply lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp:140]: Yes this is the way, thanks!

**Review by odjuricicTT — CHANGES_REQUESTED** [review]
@mbezuljTT As discussed offline, i think that the only thing that is missing for us to try this out on mnist is to somehow fill matmul program config on your side and remove it from the api.

Is this possible by setting some params to default values and inferring others?

**Review by odjuricicTT — APPROVED** [review]

**Review by odjuricicTT — APPROVED** [review]
As discussed, required changed coming in a followup PR.

**Comment by mbezuljTT** [comment]
> @mbezuljTT As discussed offline, i think that the only thing that is missing for us to try this out on mnist is to somehow fill matmul program config on your side and remove it from the api.
> 
> Is this possible by setting some params to default values and inferring others?

as discussed offline, we'll address this in following PR

**Pull request merged**